packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboard.ts

```diff
@@ -92,12 +100,9 @@ export const useGridKeyboard = (
 
       const nextCellIndexes = apiRef.current.getState().keyboard.cell!;
       // We select the rows in between
-      const rowIds = Array(Math.abs(nextCellIndexes.rowIndex - selectionFromRowIndex) + 1)
-        .fill(
-          nextCellIndexes.rowIndex > selectionFromRowIndex
-            ? selectionFromRowIndex
-            : nextCellIndexes.rowIndex,
-        )
+      const nextRowIndex = nextCellIndexes.rowIndex === null ? 0 : params.rowIndex;
```

Comment:
This should not change as you should never select headers...
Please refactor, to separate the concern for headers...

Reply:
But why, I was left with the impression that you should be able to "select" headers. Basically, if you are on a cell with coordinates 0:0 pressing the up arrow should select the header above it.

Reply:
reverted